### DISCUSSION TITLE
can ${define VAR} generate definitions without quotes?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1693] [Q&A] [answered]
目前VAR是一个string的时候，add_configfiles("xx.h.in")生成自带引号：
```
#define VAR "value"
// #define VAR "" even if VAR is empty string!
```
这样就无法实现定义/不定义的切换了；有没有办法让语句的转换不要自带引号？

**Comment by waruqi** [comment]
> 有没有办法让语句的转换不要自带引号？

目前不支持，得改

> 这样就无法实现定义/不定义的切换了；

这跟切换到不定义没啥关系吧，你不设置变量，默认就是 `#undef VAR` 就是不定义 
https://github.com/xmake-io/xmake/blob/3a8aab03b9e615cd0a40a1ed21e9ab4490f6f1c5/xmake/actions/config/configfiles.lua#L240-L242

**Comment by xq114** [comment, marked as answer]
https://github.com/xmake-io/xmake/issues/1694
